src/apps/eos/get_public_key.py

```diff
@@ -0,0 +1,40 @@
+from trezor import wire
+from trezor.crypto import base58
+from trezor.crypto.curve import secp256k1
+from trezor.crypto.hashlib import ripemd160
+from trezor.messages.EosGetPublicKey import EosGetPublicKey
+from trezor.messages.EosPublicKey import EosPublicKey
+
+from apps.common import seed
+from apps.eos.layout import require_get_public_key
+
+
+def _ripemd160_32(data: bytes) -> bytes:
+    return ripemd160(data).digest()[:4]
+
+
+def _public_key_to_wif(pub_key: bytes) -> str:
+    if len(pub_key) == 65:
+        head = 0x03 if (pub_key[64] & 0x01) == 1 else 0x02
+        compresed_pub_key = bytes([head]) + pub_key[1:33]
+    elif len(pub_key) == 33:
+        compresed_pub_key = pub_key
+    else:
+        raise wire.DataError("invalid public key length")
+    return "EOS" + base58.encode_check(compresed_pub_key, _ripemd160_32)
+
+
+def _get_public_key(node):
+    seckey = node.private_key()
+    public_key = secp256k1.publickey(seckey, True)
+    wif = _public_key_to_wif(public_key)
+    return wif, public_key
+
+
+async def get_public_key(ctx, msg: EosGetPublicKey):
+    address_n = msg.address_n or ()
```

Comment:
this should use address validation, see e.g. ethereum:
https://github.com/trezor/trezor-core/blob/master/src/apps/ethereum/get_address.py#L12
https://github.com/trezor/trezor-core/blob/master/src/apps/ethereum/address.py#L38

Reply:
I have fetched the latest version of the source code, and it seems that the node retrieval was changed. Ethereum code is not working. As i understand it is due to mnemonic module introduction. What is the current correct way to retrieve node?

Reply:
> this should use address validation, see e.g. ethereum:
> https://github.com/trezor/trezor-core/blob/master/src/apps/ethereum/get_address.py#L12
> https://github.com/trezor/trezor-core/blob/master/src/apps/ethereum/address.py#L38

Done.

Reply:
I don't encounter problems with your current code, when using the correct path prefix (e.g. m/44'/194')

are you running into something specific? maybe a full rebuild + storage erase (/var/tmp/trezor.flash) would help?

Reply:
Thanks. Those steps helped. So we can mark as resolved